lib/src/google_fonts_base.dart

```diff
@@ -181,12 +181,13 @@ Future<void> _loadFontByteData(
   String familyWithVariantString,
   Future<ByteData?>? byteData,
 ) async {
-  final anyFontDataFound = byteData != null && await byteData != null;
-  if (anyFontDataFound) {
-    final fontLoader = FontLoader(familyWithVariantString);
-    fontLoader.addFont(byteData as Future<ByteData>);
-    await fontLoader.load();
-  }
+  if (byteData == null) return;
+  final fontData = await byteData;
+  if (fontData == null) return;
+
+  final fontLoader = FontLoader(familyWithVariantString);
+  fontLoader.addFont(byteData as Future<ByteData>);
```

Comment:
Should it be like this?
```dart
  fontLoader.addFont(Future.value(fontData));
```

Reply:
I don't think so. What does this get us?

Reply:
https://github.com/material-foundation/google-fonts-flutter/issues/146#issuecomment-782176828

Because `addFont` requires a `Future<ByteData>` as input, and the value is already loaded, we can simply pack it inside a `Future.value` and pass it in.

Reply:
Could it take `FutureOr<ByteData>`? Avoiding an unnecessary wrap would be nice.

Reply:
Also, it seems this case isn't covered in tests, right?

Reply:
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/font_loader.dart#L36-L43

Unfortunately it takes only `Future<ByteData>`

Reply:
The current impl just reuses the existing future, right? That should be fine – instead of creating a new one.

Reply:
Exactly, the current uses the existing future.

Reply:
@gaetschwartz I think you'll need to wrap the value. So you have a `Future<ByteData>` instead of `Future<ByteData?>`

Have you tested this workflow to make sure it works? Are there tests?

Reply:
I ran the examples and it worked without but I can still do if you want